pkg/util/annotation/annotation.go

```diff
@@ -30,81 +29,71 @@ const (
 )
 
 // HasAnnotationKey returns true if the given object has the given annotation key in its ObjectMeta.
-func HasAnnotationKey(obj metav1.Object, key string) (bool, error) {
+func HasAnnotationKey(obj metav1.Object, key string) bool {
 	if IsNilPointer(obj) {
-		return false, fmt.Errorf("object(%T) is nil pointer", obj)
+		return false
 	}
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
-		return false, nil
+		return false
 	}
 	_, ok := annotations[key]
-	return ok, nil
+	return ok
 }
 
 // HasAnnotationKeyValue returns true if the given object has the given annotation key and value in its ObjectMeta.
-func HasAnnotationKeyValue(obj metav1.Object, key, value string) (bool, error) {
+func HasAnnotationKeyValue(obj metav1.Object, key, value string) bool {
 	if IsNilPointer(obj) {
-		return false, fmt.Errorf("object(%T) is nil pointer", obj)
+		return false
 	}
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
-		return false, nil
+		return false
 	}
 	val, ok := annotations[key]
-	return ok && value == val, nil
+	return ok && value == val
 }
 
 // AddAnnotation adds the given annotation key and value to the given objects ObjectMeta,
 // and overwrites the annotation value if it already exists.
 // Returns true if the object was updated.
-func AddAnnotation(obj metav1.Object, key, value string) (bool, error) {
-	if IsNilPointer(obj) {
-		return false, fmt.Errorf("object(%T) is nil pointer", obj)
-	}
-
-	if key == "" {
-		return false, fmt.Errorf("key is a empty string.")
+func AddAnnotation(obj metav1.Object, key, value string) bool {
+	if IsNilPointer(obj) || key == "" {
```

Comment:
The key is usually a constant, so the chance that it's empty is slim. Let's remove this `key` check to make it clear that it's the caller's responsibility to pass in a valid key.

Reply:
Removed

Reply:
Pls fix the unit tests.